### ISSUE TITLE
Favicons from wrong sites are shown for discarded tabs, posing privacy & security questions.

### ISSUE THREAD

**Issue by jarredou** [mbnuqw/sidebery#1090]
## Steps to reproduce
- Open Firefox in [new profile](https://support.mozilla.org/en-US/kb/profile-manager-create-remove-switch-firefox-profiles)
- Install Sidebery
- load tabs from different & unrelated website.
- discard/unload them.
- restart firefox
- favicons are associated with wrong sites.

## Expected behavior
Not mixing favicons with totally unrelated site.

## Actual behavior
it's mixing favicons with totally unrelated site

## Debug info

<details>
<summary><b>Addon data</b></summary>
```
{
  "addonVersion": "5.0.0b33",
  "firefoxVersion": "115.0",
  "settings": {
    "nativeScrollbars": true,
    "nativeScrollbarsThin": true,
    "nativeScrollbarsLeft": true,
    "selWinScreenshots": false,
    "updateSidebarTitle": true,
    "markWindow": false,
    "markWindowPreface": "[Sidebery] ",
    "ctxMenuNative": true,
    "ctxMenuRenderInact": true,
    "ctxMenuRenderIcons": true,
    "ctxMenuIgnoreContainers": "",
    "navBarLayout": "horizontal",
    "navBarInline": true,
    "navBarSide": "left",
    "hideAddBtn": false,
    "hideSettingsBtn": false,
    "navBtnCount": true,
    "hideEmptyPanels": true,
    "hideDiscardedTabPanels": false,
    "navActTabsPanelLeftClickAction": "none",
    "navActBookmarksPanelLeftClickAction": "none",
    "navTabsPanelMidClickAction": "discard",
    "navBookmarksPanelMidClickAction": "none",
    "navSwitchPanelsWheel": false,
    "subPanelRecentlyClosedBar": false,
    "subPanelBookmarks": false,
    "subPanelHistory": false,
    "groupLayout": "grid",
    "skipEmptyPanels": false,
    "dndTabAct": true,
    "dndTabActDelay": 750,
    "dndTabActMod": "none",
    "dndExp": "pointer",
    "dndExpDelay": 0,
    "dndExpMod": "none",
    "dndOutside": "win",
    "dndActTabFromLink": true,
    "dndActSearchTab": true,
    "dndMoveTabs": false,
    "dndMoveBookmarks": false,
    "searchBarMode": "dynamic",
    "warnOnMultiTabClose": "any",
    "activateLastTabOnPanelSwitching": true,
    "activateLastTabOnPanelSwitchingLoadedOnly": true,
    "switchPanelAfterSwitchingTab": "always",
    "tabRmBtn": "hover",
    "hideInact": false,
    "activateAfterClosing": "prev_act",
    "activateAfterClosingStayInPanel": false,
    "activateAfterClosingGlobal": true,
    "activateAfterClosingNoFolded": true,
    "activateAfterClosingNoDiscarded": true,
    "askNewBookmarkPlace": false,
    "tabsRmUndoNote": true,
    "nativeHighlight": false,
    "tabsUnreadMark": true,
    "tabsUpdateMark": "all",
    "tabsUpdateMarkFirst": true,
    "tabsReloadLimit": 5,
    "tabsReloadLimitNotif": true,
    "showNewTabBtns": true,
    "newTabBarPosition": "bottom",
    "tabsPanelSwitchActMove": true,
    "tabsPanelSwitchActMoveAuto": true,
    "tabsUrlInTooltip": "full",
    "openSubPanelOnMouseHover": false,
    "selectActiveTabFirst": true,
    "newTabCtxReopen": false,
    "moveNewTabPin": "start",
    "moveNewTabParent": "last_child",
    "moveNewTabParentActPanel": false,
    "moveNewTab": "end",
    "moveNewTabActivePin": "start",
    "pinnedTabsPosition": "top",
    "pinnedTabsList": false,
    "pinnedAutoGroup": false,
    "tabsTree": true,
    "groupOnOpen": true,
    "tabsTreeLimit": "none",
    "hideFoldedTabs": false,
    "hideFoldedParent": "none",
    "autoFoldTabs": false,
    "autoFoldTabsExcept": "none",
    "autoExpandTabs": false,
    "autoExpandTabsOnNew": false,
    "rmChildTabs": "folded",
    "tabsChildCount": true,
    "tabsLvlDots": true,
    "discardFolded": false,
    "discardFoldedDelay": 0,
    "discardFoldedDelayUnit": "sec",
    "tabsTreeBookmarks": true,
    "treeRmOutdent": "first_child",
    "ignoreFoldedParent": false,
    "colorizeTabs": false,
    "colorizeTabsSrc": "domain",
    "colorizeTabsBranches": false,
    "colorizeTabsBranchesSrc": "domain",
    "inheritCustomColor": true,
    "warnOnMultiBookmarkDelete": "collapsed",
    "autoCloseBookmarks": false,
    "autoRemoveOther": false,
    "highlightOpenBookmarks": false,
    "activateOpenBookmarkTab": false,
    "showBookmarkLen": true,
    "bookmarksRmUndoNote": true,
    "loadBookmarksOnDemand": true,
    "pinOpenedBookmarksFolder": true,
    "oldBookmarksAfterSave": "ask",
    "loadHistoryOnDemand": true,
    "fontSize": "xxs",
    "animations": true,
    "animationSpeed": "fast",
    "theme": "proton",
    "density": "compact",
    "colorScheme": "ff",
    "sidebarCSS": true,
    "groupCSS": false,
    "snapNotify": false,
    "snapExcludePrivate": false,
    "snapInterval": 1,
    "snapIntervalUnit": "hr",
    "snapLimit": 10,
    "snapLimitUnit": "snap",
    "snapAutoExport": false,
    "snapAutoExportType": "json",
    "snapAutoExportPath": "Sidebery/snapshot-%Y.%M.%D-%h.%m.%s",
    "snapMdFullTree": false,
    "hScrollAction": "none",
    "navSwitchPanelsDelay": 128,
    "scrollThroughTabs": "none",
    "scrollThroughVisibleTabs": false,
    "scrollThroughTabsSkipDiscarded": false,
    "scrollThroughTabsExceptOverflow": true,
    "scrollThroughTabsCyclic": false,
    "scrollThroughTabsScrollArea": 0,
    "autoMenuMultiSel": true,
    "multipleMiddleClose": false,
    "longClickDelay": 500,
    "wheelThreshold": false,
    "wheelThresholdX": 10,
    "wheelThresholdY": 60,
    "tabDoubleClick": "none",
    "tabsSecondClickActPrev": false,
    "tabsSecondClickActPrevPanelOnly": false,
    "shiftSelAct": true,
    "activateOnMouseUp": true,
    "tabLongLeftClick": "none",
    "tabLongRightClick": "none",
    "tabCloseMiddleClick": "close",
    "tabsPanelLeftClickAction": "none",
    "tabsPanelDoubleClickAction": "tab",
    "tabsPanelRightClickAction": "menu",
    "tabsPanelMiddleClickAction": "tab",
    "newTabMiddleClickAction": "new_child",
    "bookmarksLeftClickAction": "open_in_act",
    "bookmarksLeftClickActivate": false,
    "bookmarksLeftClickPos": "default",
    "bookmarksMidClickAction": "open_in_new",
    "bookmarksMidClickActivate": false,
    "bookmarksMidClickRemove": false,
    "bookmarksMidClickPos": "default",
    "syncName": "rigo-pc",
    "syncSaveSettings": true,
    "syncSaveCtxMenu": true,
    "syncSaveStyles": true,
    "syncSaveKeybindings": false
  },
  "permissions": {
    "allUrls": false,
    "webRequest": false,
    "webRequestBlocking": false,
    "proxy": false,
    "tabHide": false,
    "clipboardWrite": false,
    "history": false,
    "bookmarks": true,
    "downloads": false
  },
  "storage": {
    "size": "1.3 mb",
    "props": {
      "containers": "1.09 kb",
      "contextMenu": "596 b",
      "favDomains": "31.7 kb",
      "favHashes": "2.68 kb",
      "favicons": "185 kb",
      "groupCSS": "551 b",
      "lastSnapTime": "13 b",
      "panelIndex": "1 b",
      "profileID": "14 b",
      "settings": "4.4 kb",
      "sidebar": "679 b",
      "sidebarCSS": "402 b",
      "snapshots": "964 kb",
      "tabsDataCache": "141 kb",
      "ver": "10 b"
    }
  },
  "sidebar": {
    "nav": [
      "bookmarks",
      "firefox-default",
      "sp-0",
      "settings"
    ],
    "panels": {
      "bookmarks": {
        "type": 1,
        "id": "bookmarks",
        "name": "len: 9",
        "iconSVG": "icon_bookmarks",
        "iconIMGSrc": "",
        "iconIMG": "",
        "color": "toolbar",
        "lockedPanel": false,
        "tempMode": false,
        "skipOnSwitching": false,
        "rootId": "root________",
        "viewMode": "tree",
        "autoConvert": false,
        "srcPanelConfig": null
      },
      "firefox-default": {
        "type": 2,
        "id": "firefox-default",
        "name": "len: 7",
        "color": "toolbar",
        "iconSVG": "icon_tabs",
        "iconIMGSrc": "",
        "iconIMG": "",
        "lockedPanel": false,
        "skipOnSwitching": false,
        "noEmpty": false,
        "newTabCtx": "none",
        "dropTabCtx": "none",
        "moveRules": [],
        "moveExcludedTo": -1,
        "bookmarksFolderId": -1,
        "newTabBtns": [],
        "srcPanelConfig": null
      }
    }
  },
  "containers": [
    {
      "id": "firefox-container-1",
      "cookieStoreId": "firefox-container-1",
      "name": "9",
      "icon": "...",
      "color": "blue",
      "colorCode": "#37adff",
      "proxified": false,
      "proxy": null,
      "reopenRulesActive": false,
      "reopenRules": [],
      "userAgentActive": false,
      "userAgent": ""
    },
    {
      "id": "firefox-container-2",
      "cookieStoreId": "firefox-container-2",
      "name": "13",
      "icon": "...",
      "color": "orange",
      "colorCode": "#37adff",
      "proxified": false,
      "proxy": null,
      "reopenRulesActive": false,
      "reopenRules": [],
      "userAgentActive": false,
      "userAgent": ""
    },
    {
      "id": "firefox-container-3",
      "cookieStoreId": "firefox-container-3",
      "name": "8",
      "icon": "...",
      "color": "green",
      "colorCode": "#37adff",
      "proxified": false,
      "proxy": null,
      "reopenRulesActive": false,
      "reopenRules": [],
      "userAgentActive": false,
      "userAgent": ""
    },
    {
      "id": "firefox-container-4",
      "cookieStoreId": "firefox-container-4",
      "name": "15",
      "icon": "...",
      "color": "pink",
      "colorCode": "#37adff",
      "proxified": false,
      "proxy": null,
      "reopenRulesActive": false,
      "reopenRules": [],
      "userAgentActive": false,
      "userAgent": ""
    }
  ],
  "sidebarCSSLen": "378",
  "groupCSSLen": "529",
  "windows": [
    {
      "state": "maximized",
      "incognito": false,
      "tabsCount": 996
    }
  ],
  "bookmarks": {
    "bookmarksCount": 1931,
    "foldersCount": 42,
    "separatorsCount": 3,
    "maxDepth": 3
  }
}
```
</details>

![image](https://github.com/mbnuqw/sidebery/assets/6044199/81782c4c-cfab-47df-bf92-b536fac4d3ae)

Favicons with big "M" are from newspaper "lemonde.fr" and are totally unrelated with the site from these 2 tabs are from from an audio/music community forum. There's absolutely no reason to mix their favicons, and that poses privacy & security questions. Which other data the addon is mixing and associating with wrong are unralted websites ?
THIS SHOULDN'T HAPPEN.

**Comment by akhiljalagam** [comment]
I would like to inform you that I experienced a similar issue when I attempted to restore snapshots.

**Comment by mbnuqw** [comment]
Resolved in v5.0.0. Unfortunately, updating addon doesn't fix wrong favicons and you have two options:
- You can wait and let Sidebery gradually rewrite favicons (it may take months)
- You can reset favicons cache now:
  - Go to Sidebery settings >> General >> Sidebery's data
  - Delete `favicons`, `favicons_01`, `favicons_02`, `favicons_03`, `favicons_04`, `favicons_05`, `favDomains`, `favHashes` props
  - Reload addon (Sidebery settings >> Help)
